api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java

```diff
@@ -32,6 +32,8 @@
  */
 public class NoneShardSpec implements ShardSpec
 {
+  public static NoneShardSpec INSTANCE = new NoneShardSpec();
```

Comment:
I think we can also make Jackson use this by putting `@JsonCreator` on a method like `public static NoneShardSpec instance() { return INSTANCE; }`